activesupport/lib/active_support/core_ext/range/compare_range.rb

```diff
@@ -15,6 +15,7 @@ module CompareWithRange
     # The given range must be fully bounded, with both start and end.
     def ===(value)
       if value.is_a?(::Range)
+        return false if value.begin && value.end && value.begin > value.end
```

Comment:
There's one more weird case, an exclusive range where `begin == end`:

```bash
$ ruby -e "puts (1..3).cover?(2...2)"
false
```

Ruby treats this as a backwards range:

https://github.com/ruby/ruby/blob/v2_7_0/range.c#L1547

```suggestion
        return false if value.begin && value.end && value.exclude_end? ? value.begin >= value.end : value.begin > value.end
```

Could you add tests for this case too?

Reply:
Updated